src/uag.c

```diff
@@ -757,6 +758,9 @@ int uag_reset_transp(bool reg, bool reinvite)
 					ua_event(ua, UA_EVENT_CALL_CLOSED,
 						 call, "Transport of "
 						 "User Agent changed");
+					lec_prev = lec->prev ? lec->prev : lec;
+					mem_deref(call);
+					lec = lec->next ? lec : lec_prev;
```

Comment:
The common approach is to use a while loop, if destructor is used within `list` foreach itself:
```c
while (le) {
  call = lec->data;
  lec = lec->next;
  ...
}
```